src/backend/src/services/work-packages.services.ts

```diff
@@ -701,4 +702,87 @@ export default class WorkPackagesService {
 
     return workPackageTemplateTransformer(workPackage);
   }
+
+  static async editWorkPackageTemplate(
+    submitter: User,
+    workPackageTemplateId: string,
+    templateName: string,
+    templateNotes: string,
+    duration: number | undefined,
+    stage: WorkPackageStage | undefined,
+    blockedBy: BlockedByCreateArgs[],
+    expectedActivities: string[],
+    deliverables: string[],
+    workPackageName: string | undefined
+  ): Promise<WorkPackageTemplate> {
+    const originalWorkPackageTemplate = await prisma.work_Package_Template.findUnique({
+      where: { workPackageTemplateId },
+      include: { blockedBy: true }
+    });
+
+    if (!originalWorkPackageTemplate) throw new NotFoundException('Work Package Template', workPackageTemplateId);
+    if (originalWorkPackageTemplate.dateDeleted) throw new DeletedException('Work Package Template', workPackageTemplateId);
+    if (!isAdmin(submitter.role)) throw new AccessDeniedAdminOnlyException('edit work package templates');
+
+    const updatedBlockedByIds = blockedBy.map((item) => item.blockedByInfoId);
+
+    // deleting a blocked by if the new list does not contain it
```

Comment:
can you add a comment like this to each list of Ids to specify how you're deciding which ones to create, update, delete